pkg/rotator/rotator.go

```diff
@@ -752,6 +760,10 @@ func (r *ReconcileWH) Reconcile(ctx context.Context, request reconcile.Request)
 		return reconcile.Result{}, nil
 	}
 
+	if r.enableReadinessCheck {
+		<-r.certsMounted
```

Comment:
Thanks for the PR @p-strusiewiczsurmacki-mobica !

This change in the Reconcile function and the readiness check mechanism using channel certsMounted can potentially lead to blocking behavior in the Reconcile function if the `ensureCertsMounted` function fails to close certsMounted. Instead, we can add a retry mechanism to handle failure cases.

```suggestion
 select {
    case <-r.certsMounted:
        // Continue with reconciliation
    case <-r.certsNotMounted:
        // Requeue with backoff strategy
        return reconcile.Result{Requeue: true, RequeueAfter: 5 * time.Second}, errors.New("certs not mounted, retrying reconciliation")
    }
```

Reply:
Thank you! That's cleaver! Didn't think of that, tbh.

Reply:
In the `<-r.certsNotMounted` case I would `return reconcile.Result{}, errors.New("certs not mounted, retrying reconciliation")`.  For completeness I would also check `ctx.Done`.

Reply:
After some consideration I think need to think this one more thoroughly. If `certsNotMounted` is closed, then Rotator emits error and quits I believe. So there might be no point in waiting for it.

Reply:
I've changed the approach to use context's timeout for check. Please let me know what do you think about this one.

Reply:
@p-strusiewiczsurmacki-mobica I think your previous implementation is cleaner if you include the suggestion here with this update

```go
 select {
    case <-r.certsMounted:
        // Continue with reconciliation
    case <- ctx.Done():
       // Continue with reconciliation
    case <-r.certsNotMounted:
        // controller-runtime will requeue with backoff strategy when this error is returned
        return reconcile.Result{}, errors.New("certs not mounted, retrying reconciliation")
  } 
```

Per controller-runtime:
> // If the returned error is non-nil, the Result is ignored and the request will be
	// requeued using exponential backoff.

https://github.com/kubernetes-sigs/controller-runtime/blob/d8b679322d5cc466d12301b8875c42040fd29e46/pkg/reconcile/reconcile.go#L112-L114

Reply:
@ritazh OK, I've updated the code.